visualization/app/codeCharta/ui/codeMap/codeMap.preRender.service.ts

```diff
@@ -110,11 +110,8 @@ export class CodeMapPreRenderService implements StoreSubscriber, MetricServiceSu
 	}
 
 	private getSelectedFilesAsUnifiedMap(): CCFile {
-		const fileStates: FileState[] = this.fileStateService.getFileStates()
+		const fileStates: FileState[] = _.cloneDeep(this.fileStateService.getFileStates())
```

Comment:
why do we clone here?

Reply:
there was a reason. Is there some useful commit message?

Reply:
Well, we mutate the FileState that was passed by reference here. I guess whoever implemented this did not realize that this will also change the object in the FileStateService. Until this PR it did not really matter that we did that, since the map was redecorated completely anyways. However, if we predecorate the Paths earlier, it starts to matter. What happens without the deepClone can be seen in my first comment to this PR.

Reply:
Alright. We should address this issue later. I think it makes sense to not return value from the decoration-methods and just mutate the parameter indicating that there are obviously side effects to it.

For now, let's keep the cloneDeep. Be careful with that in production code though. :)